pkg/services/ngalert/schedule/fetcher.go

```diff
@@ -16,10 +19,21 @@ func (sch *schedule) getAlertRules(ctx context.Context, disabledOrgs []int64) []
 	q := models.GetAlertRulesForSchedulingQuery{
 		ExcludeOrgIDs: disabledOrgs,
 	}
-	err := sch.ruleStore.GetAlertRulesForScheduling(ctx, &q)
-	if err != nil {
-		sch.log.Error("failed to fetch alert definitions", "err", err)
-		return nil
+
+	// If the database is unavailable or the query returns an error then return
+	// the alert rules from the most recent tick
+	if err := sch.ruleStore.GetAlertRulesForScheduling(ctx, &q); err != nil {
+		sch.log.Error("failed to get most recent alert rules", "error", err)
```

Comment:
How do you feel about including the number of alerts we're about to return from the cache?